packages/date-picker/src/DatePicker/DatePicker.tsx

```diff
@@ -33,10 +29,10 @@ export interface DatePickerProps {
   initialMonth?: Date
 
   // The date passed in from the consumer that renders in the input and calendar.
-  value?: Date
+  valueDate: Date | undefined
 
-  // Event passed from consumer to handle the date on change.
-  onChange: (day: Date) => void
+  // Setter for updating the selected date used by the consumer.
+  setValueDate: React.Dispatch<React.SetStateAction<Date | undefined>>
```

Comment:
🤔 This seems a bit odd to limit this to a react set state action.

The usual way we do this was how it was before, curious what drove the change here.